File: src/test/java/org/goafabric/calleeservice/adapter/CalleeServiceAdapterIT.java
package org.goafabric.calleeservice.adapter;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class CalleeServiceAdapterIT {
    @Autowired
    private CalleeServiceAdapter calleeServiceAdapterDec;

    @Test
    void sayMyName() {
        Assertions.assertThat(calleeServiceAdapterDec.sayMyName("Heisenberg").message())
                .isEqualTo("Your name is: Heisenberg");
    }

    @Test
    void sayMyOtherName() {
        Assertions.assertThat(calleeServiceAdapterDec.sayMyOtherName("SlimShady").message())
                .isEqualTo("Your other name is: SlimShady");
    }
}

File: src/test/kotlin/org/goafabric/calleeservice/adapter/CalleeServiceAdapterIT.kt
package org.goafabric.calleeservice.adapter

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
internal class CalleeServiceAdapterIT(
    @Autowired private val calleeServiceAdapter: CalleeServiceAdapter) {

    @Test
    fun sayMyName() {
        assertThat(calleeServiceAdapter.sayMyName("Heisenberg")!!.message)
            .isEqualTo("Your name is: Heisenberg")
    }

    @Test
    fun sayMyOtherName() {
        assertThat(calleeServiceAdapter.sayMyName("SlimShady")!!.message)
            .isEqualTo("Your name is: SlimShady")
    }
}